front/src/pages/course-selection/creacion/form-institucion.js

```diff
@@ -0,0 +1,201 @@
+import { Formik, Form, Field } from 'formik';
+
+class FormInstitucion extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      items: [],
+      isEmpty: false,
+      institucionOptions: TIPOS_INSTITUCION,
+      nivelesOptions: NIVELES,
+      selectedInst: '',
+      selectedNiveles: [],
+      nombre: '',
+      telefono: 0,
+    };
+  }
+
+  handleChange = (event) => {
+    const { value, name } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleInstChange = (selectedInst) => {
+    if (selectedInst) {
+      this.setState({ selectedInst });
+    } else {
+      this.setState({
+        selectedInst: '',
+      });
+    }
+  };
+
+  handleNivelesChange = (selectedNiveles) => {
+    this.setState({ selectedNiveles });
+  };
+
+  onUserSubmit = async () => {
+    const nivelesState = [...this.state.selectedNiveles];
+    const nivelesArray = nivelesState.map((elem) => elem.value);
+
+    const obj = {
+      niveles: nivelesArray,
+      nombre: this.state.nombre,
+      telefono: this.state.telefono,
+      tipo: this.state.selectedInst.value,
+    };
+    const instRef = await addDocument(
+      'instituciones',
+      obj,
+      this.props.user,
+      'Institucion creada!',
+      'Institución creada con éxito',
+      'Error al crear la Institución'
+    );
+    localStorage.setItem(
+      'institution',
+      JSON.stringify({ id: instRef.id, name: obj.nombre })
+    );
+    this.props.history.push({
+      pathname: '/seleccion-curso/crear-curso',
+      instId: instRef.id,
+    });
+  };
+
+  render() {
+    const {
+      selectedInst,
+      institucionOptions,
+      selectedNiveles,
+      nivelesOptions,
+    } = this.state;
+    return (
+      <Row className="h-100">
+        <Colxx xxs="12" md="10" className="mx-auto my-auto">
+          <Card className="auth-card">
+            <div className="form-side-full">
+              <NavLink to={`/`} className="white">
+                <span className="logo-single" />
+              </NavLink>
+              <CardTitle className="mb-4">
+                <IntlMessages id="institucion.complete-datos" />
+              </CardTitle>
+              <Formik onSubmit={this.onUserSubmit}>
+                {({ errors, touched }) => (
+                  <Form className="av-tooltip tooltip-label-bottom">
+                    <FormGroup className="form-group has-float-label">
+                      <Label>
+                        <IntlMessages id="institucion.nombre" />
+                      </Label>
+                      <Field
+                        className="form-control"
+                        name="nombre"
+                        onChange={this.handleChange}
+                      />
+                      {errors.nombre && touched.nombre && (
+                        <div className="invalid-feedback d-block">
+                          {errors.nombre}
+                        </div>
+                      )}
```

Comment:
Esto es una validación del nombre? Que validaria?

Reply:
Actualmente no está validando nada porque no tiene método de validación. Para ver un ejemplo, en login.js está el validate={this.validateEmail}

Reply:
Excelenchi